src/js/apps/patients/worklist/worklist-owned-by_app.js

```diff
@@ -0,0 +1,194 @@
+        );
+      }
+      return dateSort('asc', a.model.get('due_date'), b.model.get('due_date'));
+    },
+  },
+  {
+    id: 'sortDueDesc',
+    text: i18n.sortOptions.sortDueDesc,
+    comparator(a, b) {
+      if (a.model.get('due_date') === b.model.get('due_date')) {
+        return dateSort(
+          'desc',
+          moment(a.model.get('due_time'), 'HH:mm:ss'),
+          moment(b.model.get('due_time'), 'HH:mm:ss'),
+        );
+      }
+      return dateSort('desc', a.model.get('due_date'), b.model.get('due_date'));
+    },
+  },
+  {
+    id: 'sortUpdateAsc',
+    text: i18n.sortOptions.sortUpdateAsc,
+    comparator(a, b) {
+      return dateSort('asc', a.model.get('updated_at'), b.model.get('updated_at'));
+    },
+  },
+  {
+    id: 'sortUpdateDesc',
+    text: i18n.sortOptions.sortUpdateDesc,
+    comparator(a, b) {
+      return dateSort('desc', a.model.get('updated_at'), b.model.get('updated_at'));
+    },
+  },
+]);
+
+export default App.extend({
+  childApps: {
+    filters: FiltersApp,
+  },
+  stateEvents: {
+    'change': 'onChangeState',
+  },
+  onChangeState(state) {
+    store.set(this.stateId, state);
+    if (state.hasChanged('groupId') || state.hasChanged('clinicianId')) {
+      this.restart();
+    }
+  },
+  viewEvents: {
+    'toggle:listType': 'onToggleListType',
+  },
+  initListState() {
+    this.stateId = `worklist-${ this.worklistId }-${ this.worklistType }`;
+
+    const default_state = {
+      sortId: 'sortUpdateDesc',
+      groupId: 'all-groups',
+      clinicianId: this.currentClinician.id,
+    };
+
+    if (!store.get(this.stateId)) {
+      store.set(this.stateId, default_state);
+      this.setState(default_state);
+    } else {
+      this.setState(store.get(this.stateId));
+    }
+  },
+  onBeforeStart({ worklistId, worklistType }) {
+    if (this.isRestarting()) {
+      this.getRegion('list').startPreloader();
+      return;
+    }
+
+    this.worklistId = worklistId;
+    this.worklistType = worklistType;
+    this.currentClinician = Radio.request('bootstrap', 'currentUser');
+    this.groups = this.currentClinician.getGroups();
+
+    this.initListState();
+
+    this.showView(new LayoutView({
+      worklistId: this.worklistId,
+      isFlowList: this.worklistType === 'flows',
+    }));
+
+    this.getRegion('list').startPreloader();
+    this.showSortView();
+  },
+  beforeStart() {
+    let groupId = this.getState('groupId');
+
+    if (groupId === 'all-groups') {
+      groupId = this.groups.pluck('id').join(',');
+    }
+
+    const filter = _.extend({ group: groupId }, this._filtersById(this.worklistId, this.getState('clinicianId')));
+
+    const filtersApp = this.startChildApp('filters', {
+      region: this.getRegion('filters'),
+      clinician: this.currentClinician,
+      groupId: this.getState('groupId') || this.groups.at(0).id,
+      clinicianId: this.getState('clinicianId') || this.currentClinician.id,
+    });
+
+    this.listenTo(filtersApp, {
```

Comment:
you _could_ simplify this a bit I think.. I don't think it'd be too brittle, but this could be:
```js
this.listenTo(filtersApp, 'change:state', state = > {
  this.setState(state);
});
```
then in the change event's you'll be triggering `{ groupId: id }` etc